Add magit-delta recipe

dandavison opened the pull request:
This PR adds [magit-delta](https://github.com/dandavison/magit-delta) to MELPA.

`magit-delta` supplies a minor mode `magit-delta-mode` which integrates [Delta](https://github.com/dandavison/delta) with Magit, so that diffs in Magit are displayed with color syntax highlighting provided by Delta. A screenshot of the minor mode being used is below.

**There is one important outstanding issue regarding which I would like to ask for MELPA maintainers' advice**: `magit-delta` depends on a commit merged to magit master 20 days ago: https://github.com/magit/magit/commit/8de6ecf5c5c840f8a964c3e5bd4d7a1aedf04e10. What needs to happen in order for `magit-delta` to be able to declare this dependency in a way that will make it appropriate for inclusion in MELPA (unstable)?

<br>
<table><tr><td>
  <img width=500px src="https://user-images.githubusercontent.com/52205/80056404-23745500-84f2-11ea-9ecd-832376faf2f1.png" alt="image" />
</td></tr></table>


### Direct link to the package repository
https://github.com/dandavison/magit-delta

### Your association with the package

Author & maintainer

### Relevant communications with the upstream package maintainer
**None needed**

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

walseb commented:
I'm not a melpa maintainer, but this looks great! I wonder if there is a way to generalize it to work with all diffs, not just those in magit. As for your question about wanting to depend on anything past pretty much the latest commit of magit, I think you just have to wait for magit to bump the version number if they haven't and then put that version into the header of your elisp file header, just like you have done with xterm and emacs.

riscy commented:
> There is one important outstanding issue regarding which I would like to ask for MELPA maintainers' advice: magit-delta depends on a commit merged to magit master 20 days ago: magit/magit@8de6ecf. What needs to happen in order for magit-delta to be able to declare this dependency in a way that will make it appropriate for inclusion in MELPA (unstable)?

You can require on the snapshot (unstable) version of magit e.g. `(magit "20181103")` (or whatever the date happens to be).  Package-lint will warn you but it's not a blocker.

In the meantime I can also nag you about the license boilerplate (I see your repo is licensed but the file still needs boilerplate) :)
- Please add license boilerplate or an [SPDX-License-Identifier](https://spdx.org/using-spdx-license-identifier) to magit-delta.el

dandavison commented:
@riscy @walseb thanks!

I've added a dependency on magit using the date one day after the required commit was merged to magit master. I've also added the SPDX license identifier.

@walseb Interesting. We could definitely add a mode that displays diffs using delta, via xterm-color. Perhaps some further integration with vc-mode would make sense, but I don't really use vc-mode to my knowledge, so I'm not sure yet what would make sense. I think it makes sense to continue with the mode proposed here as a package narrowly focused on magit-delta integration, and leave a separate more generalized mode as a possibility for the future.

riscy commented:
Very cool!
